messaging-activemq/src/main/resources/schema/wildfly-messaging-activemq_9_0.xsd

```diff
@@ -0,0 +1,996 @@
+        <xs:attribute name="client-id" type="xs:string" use="optional" />
+        <xs:attribute name="dups-ok-batch-size" type="xs:int" use="optional" />
+        <xs:attribute name="transaction-batch-size" type="xs:int" use="optional" />
+        <xs:attribute name="block-on-acknowledge" type="xs:boolean" use="optional" />
+        <xs:attribute name="block-on-non-durable-send" type="xs:boolean" use="optional" />
+        <xs:attribute name="block-on-durable-send" type="xs:boolean" use="optional" />
+        <xs:attribute name="auto-group" type="xs:boolean" use="optional" />
+        <xs:attribute name="pre-acknowledge" type="xs:boolean" use="optional" />
+        <xs:attribute name="retry-interval" type="xs:long" use="optional" />
+        <xs:attribute name="retry-interval-multiplier" type="xs:double" use="optional" />
+        <xs:attribute name="max-retry-interval" type="xs:long" use="optional" />
+        <xs:attribute name="reconnect-attempts" type="xs:int" use="optional" />
+        <xs:attribute name="failover-on-initial-connection" type="xs:boolean" use="optional" />
+        <xs:attribute name="connection-load-balancing-policy-class-name" type="xs:string" use="optional" />
+        <xs:attribute name="use-global-pools" type="xs:boolean" use="optional" />
+        <xs:attribute name="scheduled-thread-pool-max-size" type="xs:int" use="optional" />
+        <xs:attribute name="thread-pool-max-size" type="xs:int" use="optional" />
+        <xs:attribute name="group-id" type="xs:string" use="optional" />
+        <xs:attribute name="deserialization-white-list" type="stringList" use="optional" />
+        <xs:attribute name="deserialization-black-list" type="stringList" use="optional" />
+        <xs:attribute name="initial-message-packet-size" type="xs:int" use="optional" />
+        <xs:attribute name="use-topology-for-load-balancing" type="xs:boolean" use="optional" />
+    </xs:attributeGroup>
+
+    <xs:simpleType name="connectionFactoryType">
+        <xs:restriction base="xs:string">
+            <xs:enumeration value="GENERIC"/>
+            <xs:enumeration value="TOPIC"/>
+            <xs:enumeration value="QUEUE"/>
+            <xs:enumeration value="XA_GENERIC"/>
+            <xs:enumeration value="XA_QUEUE"/>
+            <xs:enumeration value="XA_TOPIC"/>
+        </xs:restriction>
+    </xs:simpleType>
+
+    <xs:complexType name="socket-discovery-groupType">
+        <xs:attribute name="name" type="xs:string" use="required" />
+        <xs:attribute name="socket-binding" type="xs:string"  use="required" />
+        <xs:attribute name="refresh-timeout" type="xs:long" />
+        <xs:attribute name="initial-wait-timeout" type="xs:long" />
+    </xs:complexType>
+
+    <xs:complexType name="jgroup-discovery-groupType">
+        <xs:attribute name="name" type="xs:string" use="required" />
+        <xs:attribute name="jgroups-stack" type="xs:string" >
```

Comment:
do we need to add a deprecated attribute to a new element?

Reply:
yes because the old behaviour is still present